jax/_src/scipy/linalg.py

```diff
@@ -595,3 +611,48 @@ def polar(a, side='right', method='qdwh', eps=None, maxiter=50):
    unitary, posdef, _ = lax_polar.polar(a, side=side, method=method, eps=eps,
                                         maxiter=maxiter)
    return unitary, posdef
+
+@jit
+def _sqrtm_triu(T):
+  diag = jnp.sqrt(jnp.diag(T))
+  n = diag.size
+  U = jnp.diag(diag)
+
+  def i_loop(l, data):
+    j, U = data
+    i = j - 1 - l
+    s = lax.fori_loop(i + 1, j, lambda k, val: val + U[i, k] * U[k, j], 0.0)
+    value = jnp.where(T[i, j] == s, 0.0,
+                      (T[i, j] - s) / (diag[i] + diag[j]))
+    return j, U.at[i, j].set(value)
+
+  def j_loop(j, U):
+    _, U = lax.fori_loop(0, j, i_loop, (j, U))
+    return U
```

Comment:
do you have a reference for where this algorithm came from? this would be helpful to include in a comment